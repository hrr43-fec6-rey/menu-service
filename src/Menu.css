.restaurant-title {
  font-size: 48px;
  font-family: BrandonText;
  font-weight: 700;
  background-color: #fff;
  padding: 24px 0 24px 0;
  border-bottom: 1px solid #d8d9db;
}

.menu-and-nav {
  position: relative;
  max-height: 500px;
  transition: max-height 0.2s;
  overflow: hidden;
  border-bottom: 1px solid #d8d9db;
  padding-bottom: 66px;
}

.menu-main {
  color: #2d333f;
  background-color: white;
  font-family: BrandonText;
  max-width: 640px;
}

.menu-title {
  font-size: 24px;
  font-weight: bold;
  margin-bottom: 12px;
}

.menu-nav {
  display: flex;
  flex-wrap: wrap;
  border-top: 1px solid #d8d9db;
  border-bottom: 1px solid #d8d9db;
  padding-top: 6px;
  padding-bottom: 6px;
}

.menu-nav-item {
  display: inline-block;
  justify-content: center;
  align-items: center;
  font-size: 14px;
  font-weight: bold;
  padding: 9px 16px 9px 16px;
  margin: 6px 12px 6px 0;
  border: 1px solid #d8d9db;
  border-radius: 2px;
  cursor: pointer;
  box-sizing: border-box;
  text-transform: none;
  overflow: visible;
  user-select: none;
}

.menu-nav-item-selected {
  border: 2px solid #da3743;
  padding: 8px 15px 8px 15px;
  outline: none;
}

.menu-menu-description {
  font-size: 16px;
  padding-top: 16px;
  padding-bottom: 12px;
  border-bottom: 1px solid #d8d9db;
  margin-bottom: 12px;
}

.menu-section {
  padding-bottom: 10px;
  margin-bottom: 12px;
  border-bottom: 1px solid #d8d9db;
}

.menu-section-title {
  font-size: 16px;
  font-weight: bold;
  padding-bottom: 12px;
}

.menu-items {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  flex-direction: row;
  width: 100%;
}

.menu-item {
  font-size:14px;
  font-weight: normal;
  width: 47%;
  min-width: 47%;
  display: inline-block;
  padding-bottom:16px;
}

.menu-item-title {
  display: flex;
  justify-content: space-between;
  font-weight: bold;
  margin-bottom: 8px;
  padding-right: 16px;
}

.menu-item-title-name {
  display: inline-block;
}

.menu-item-description {
  padding-right: 16px;
}

.menu-collapse {
  position: absolute;
  left: 0;
  bottom: 0;
  width:100%;
}

.menu-collapse1 {
  background-image: linear-gradient(to top,#fff,rgba(255,255,255,0));
  height: 200px;
  width:100%;
}
.menu-collapse2 {
  background-color: white;
  height: 60px;
  width: 100%;
}

.container1{
  display: flex;
  align-items: flex-start;
  justify-content: center;
  width: 100%;
  height: 100%;
}

.menu-view {
  display: inline-block;
  font-size: 16px;
  font-weight: bold;
  padding: 12px 0 12px 0;
  text-align: center;
  width: 200px;
  border: 1px solid #d8d9db;
  cursor: pointer;
  user-select: none;
  box-sizing: border-box;
  outline-color: #da3743;
}

.menu-view:hover {
  /* border: 2px solid #da3743; */
  outline-width: 1px;
  outline-style: solid;
  outline-color: #da3743;
}

.full-view {
  /* height: auto; */
  max-height: 5000px;
  transition: max-height 0.5s;
}