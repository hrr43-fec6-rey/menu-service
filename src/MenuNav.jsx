/* eslint-disable react/no-array-index-key */
import React from 'react';
import PropTypes from 'prop-types';

const MenuNav = ({
  menus, menuSelected, viewFull, action, actionView,
}) => {
  function handleClick(index) {
    action(index);
  }

  function handleKeyPress(index, e) {
    if (e.key === 'Enter' || e.key === ' ') {
      e.preventDefault();
      action(index);
    }
  }

  function handleView() {
    actionView();
  }

  function handleKeyPress2(e) {
    if (e.key === 'Enter' || e.key === ' ') {
      actionView();
    }
  }

  const viewClass = viewFull ? ' full-view' : '';

  return (
    <div className={`menu-and-nav${viewClass}`}>
      <div className="menu-nav">
        {menus.map((menu, index) => (
          <div
            className={`menu-nav-item${index === menuSelected ? ' menu-nav-item-selected' : ''}`}
            key={index}
            role="button"
            tabIndex="0"
            onClick={handleClick.bind(this, index)}
            onKeyPress={handleKeyPress.bind(this, index)}
          >
            {menu.title}
          </div>
        ))}
      </div>
      <div className="menu-menu-description">
        {menus[menuSelected].description}
      </div>

      {menus[menuSelected].sections.map((section, index) => (
        <div className="menu-section" key={index}>
          <div className="menu-section-title">
            {section.title}
          </div>
          <div className="menu-items">
            {section.items.map((item, index2) => (
              <div className="menu-item" key={index2}>
                <div className="menu-item-title">
                  <div className="menu-item-title-name">
                    {item.title}
                  </div>
                  <div className="menu-item-title-price">
                    {item.price}
                  </div>
                </div>
                <div className="menu-item-description">
                  {item.description}
                </div>
              </div>
            ))}
          </div>
        </div>
      ))}
      <div className="menu-collapse">
        {viewFull ? null : <div className="menu-collapse1" />}
        <div className="menu-collapse2">
          <div className="container1">
            <div
              key="mykey"
              className="menu-view"
              role="button"
              tabIndex="0"
              onClick={handleView}
              onKeyPress={handleKeyPress2}
            >
              {viewFull ? 'Collapse menu' : 'View full menu'}
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

MenuNav.defaultProps = {
  menus: null,
  menuSelected: 0,
  viewFull: false,
  action: () => {},
  actionView: () => {},
};

MenuNav.propTypes = {
  menus: PropTypes.arrayOf(PropTypes.object),
  menuSelected: PropTypes.number,
  viewFull: PropTypes.bool,
  action: PropTypes.func,
  actionView: PropTypes.func,
};

export default MenuNav;
