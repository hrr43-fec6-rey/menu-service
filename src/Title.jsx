import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

function Title(props) {
  const [restaurantName, setRestaurantName] = useState('Loading....');

  useEffect(() => {
    const { id } = props;
    console.log(`requesting ${id}`);
    fetch(`/gettitle/${id}`)
      .then((response) => response.json())
      .then((myJson) => {
        setRestaurantName(myJson);
        console.log(`title = ${myJson}`);
      });
  }, []);

  return (
    <div className="restaurant-title">
      {restaurantName}
    </div>
  );
}

Title.defaultProps = {
  id: '1',
};

Title.propTypes = {
  id: PropTypes.string,
};

export default Title;
