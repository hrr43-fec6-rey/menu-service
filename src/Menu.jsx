import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import './Menu.css';
import MenuNav from './MenuNav';

const Menu = ({ id }) => {
  const [menus, setMenus] = useState([{ description: '', sections: [] }]);
  const [menuSelected, setMenuSelected] = useState(0);
  const [viewFull, setViewFull] = useState(false);

  useEffect(() => {
    fetch(`/getmenu/${id}`)
      .then((response) => response.json())
      .then((myJson) => {
        setMenus(myJson[0].menus);
      });
  }, []);

  function selectMenu(index) {
    setMenuSelected(index);
  }

  function changeView() {
    setViewFull(!viewFull);
  }
  return (
    <div className="menu-main">
      <div className="menu-title">Menu</div>
      <MenuNav
        menus={menus}
        menuSelected={menuSelected}
        viewFull={viewFull}
        action={selectMenu}
        actionView={changeView}
      />
    </div>
  );
};

Menu.defaultProps = {
  id: '1',
};

Menu.propTypes = {
  id: PropTypes.string,
};

export default Menu;
