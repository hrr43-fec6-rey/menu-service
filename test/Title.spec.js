import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Title from '../src/Title.jsx';
import MenuNav from '../src/MenuNav.jsx';
import Menu from '../src/Menu.jsx';

describe('Test Title component', () => {
  let wrapper = shallow(<Title />);

  it('should not return an error', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

// describe('Test MenuNav component', () => {
//   let wrapper = shallow(<MenuNav />);

//   it('should not return an error', () => {
//     expect(wrapper).toMatchSnapshot();
//   });
// });

// describe('Test Menu component', () => {
//   let wrapper = shallow(<Menu />);

//   it('should not return an error', () => {
//     expect(wrapper).toMatchSnapshot();
//   });
// });