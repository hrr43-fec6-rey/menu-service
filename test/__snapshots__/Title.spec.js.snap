// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Menu component should not return an error 1`] = `ShallowWrapper {}`;

exports[`Test MenuNav component should not return an error 1`] = `ShallowWrapper {}`;

exports[`Test Title component should not return an error 1`] = `ShallowWrapper {}`;
